torch/futures/__init__.py

```diff
@@ -136,32 +136,12 @@ def wait_all(futures):
 
 
 T = TypeVar("T")
-GenericWithOneTypeVar = Generic[T]
 
-
-try:
-
-    # Combine the implementation class and the type class.
-    class Future(_PyFuture, GenericWithOneTypeVar):
-        r"""
-        Wrapper around a ``torch._C.Future`` which encapsulates an asynchronous
-        execution of a callable, e.g. :meth:`~torch.distributed.rpc.rpc_async`. It
-        also exposes a set of APIs to add callback functions and set results.
-        """
-        pass
-
-
-except TypeError as exc:
-    # TypeError: metaclass conflict: the metaclass of a derived class
-    # must be a (non-strict) subclass of the metaclasses of all its bases
-    class FutureMeta(_PyFuture.__class__, GenericWithOneTypeVar.__class__):
-        pass
-
-    # Combine the implementation class and the type class.
-    class Future(_PyFuture, GenericWithOneTypeVar, metaclass=FutureMeta):
-        r"""
-        Wrapper around a ``torch._C.Future`` which encapsulates an asynchronous
-        execution of a callable, e.g. :meth:`~torch.distributed.rpc.rpc_async`. It
-        also exposes a set of APIs to add callback functions and set results.
-        """
-        pass
```

Comment:
Can't remove this part.

https://app.circleci.com/pipelines/github/pytorch/pytorch/191961/workflows/54e33fc0-fd01-4835-a46a-69d9ab883315/jobs/6297692/steps

Reply:
Yes, looks like this is due to the bug in Python-3.6, found a slightly better way of doing it (by using default metaclass in 3.7 plus and one that relies on GenericMeta for 3.6)